systemvm/debian/opt/cloud/bin/cs/CsNetfilter.py

```diff
@@ -189,7 +189,8 @@ def compare(self, list):
     def add_chain(self, rule):
         """ Add the given chain if it is not already present """
         if not self.has_chain(rule.get_table(), rule.get_chain()):
-            CsHelper.execute("iptables -t %s -N %s" % (rule.get_table(), rule.get_chain()))
+            if rule.get_chain() != "":
```

Comment:
so when rule.get_chain() != self.chain && rule.get_chain() != "", we still add it to self. chain?

Reply:
I found an error in logs where it was trying to add an empty chain: "iptables -t filter -N"
I did not debug further as to why an empty chain gets added to `self.chain`. However, removing the case completely (the self.chain add) created several other failures/side-effects so I kept the change minimal.

Reply:
I would use:

<pre>
if rule.get_chain():
</pre>

This would check for None and empty strings in Python